src/core/devexec.h

```diff
@@ -88,7 +89,7 @@ class CDeviceExecution {
      *
      * The element 0 is always the timer event source.
      */
-    SEventHandlerElement mRegisteredEventHandlers[cgNumberOfHandlers];
+    std::vector<SEventHandlerElement> mRegisteredEventHandlers;
```

Comment:
as we have a given fixed size we could use here either a std::array or reserve the required size in the constructor of the DeviceExecutor class.